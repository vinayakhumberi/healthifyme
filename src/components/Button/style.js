import styled from 'styled-components';
import { rem } from '../../utils/style.util';

export const Btn = styled.button`
  background-color: ${props =>
    props.isDisabled || props.isLoading ? '#ccc' : props.backgroundColor};
  border: ${props => props.border ? props.border : 'none'};
  outline: none;
  color: ${props =>
    props.isDisabled || props.isLoading ? 'white' : props.color};
  height: ${props =>
    typeof props.height === "string" ? props.height : rem(props.height)};
  width: ${props =>
    typeof props.width === "string" ? props.width : rem(props.width)};
  font-size: ${props => rem(props.fontSize)};
  font-weight: ${props => props.fontWeight};
  margin-left: ${props =>
    typeof props.marginLeft === "string"
      ? props.marginLeft
      : rem(props.marginLeft)};
  margin-right: ${props =>
    typeof props.marginRight === "string"
      ? props.marginRight
      : rem(props.marginRight)};
  margin-top: ${props =>
    typeof props.marginTop === "string"
      ? props.marginTop
      : rem(props.marginTop)};
  margin-bottom: ${props =>
    typeof props.marginBottom === "string"
      ? props.marginBottom
      : rem(props.marginBottom)};
  cursor: ${props =>
    props.isDisabled || props.isLoading ? "auto" : "pointer"};
  border-radius: 4px;
`;
