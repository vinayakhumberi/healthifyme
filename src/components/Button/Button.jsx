import React from 'react';
import PropTypes from 'prop-types';

import { Btn } from './style';

export default function Button(props) {
  return (
    <Btn
      data-test="button-container"
      backgroundColor={props.backgroundColor}
      color={props.color}
      position={props.position}
      left={props.left}
      bottom={props.bottom}
      height={props.height}
      width={props.width}
      fontWeight={props.fontWeight}
      fontSize={props.fontSize}
      marginLeft={props.marginLeft}
      marginRight={props.marginRight}
      onClick={props.isDisabled ? null : props.onClick}
      isDisabled={props.isDisabled}
      marginTop={props.marginTop}
      marginBottom={props.marginBottom}
      data-atm={props.dataAtm}
      border={props.border}
    >
      {props.buttonText}
    </Btn>
  );
}

Button.propTypes = {
  color: PropTypes.string,
  backgroundColor: PropTypes.string,
  buttonText: PropTypes.string,
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  marginTop: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  marginBotom: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  marginLeft: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  marginRight: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  fontSize: PropTypes.number,
  fontWeight: PropTypes.string,
  isDisabled: PropTypes.bool,
  onClick: PropTypes.func,
};

Button.defaultProps = {
  backgroundColor: '#ec445c',
  color: '#ffffff',
  height: 45,
  width: '100%',
  buttonText: null,
  fontSize: 14,
  fontWeight: 'normal',
  marginTop: 0,
  marginBottom: 0,
  marginLeft: 0,
  marginRight: 0,
  isDisabled: false,
  onClick: () => { }
};
