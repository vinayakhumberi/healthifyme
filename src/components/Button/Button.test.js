import React from 'react';
import sinon from 'sinon';
import { mount } from 'enzyme';
import Button from './Button';

describe('<Button />', () => {
  let wrapper;
  let onButtonClick = sinon.spy();
  beforeEach(() => wrapper = mount(<Button onClick={onButtonClick} />));
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('Should Render a Button Container', () => {
    expect(wrapper.find('button[data-test="button-container"]').length).toEqual(1);
  });
  it('Should Simulate click events', () => {
    wrapper.find('button').simulate('click');
    expect(onButtonClick.callCount).toEqual(1);
  });
});


describe('<Button isDisabled />', () => {
  let wrapper;
  let onButtonClick = sinon.spy();
  beforeEach(() => wrapper = mount(<Button isDisabled onClick={onButtonClick} />));
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('Should Render a Button Container', () => {
    expect(wrapper.find('button[data-test="button-container"]').length).toEqual(1);
  });
  it('Should pass isDisabled prop to Button as true', () => {
    expect(wrapper.find('[data-test="button-container"]').get(0).props.isDisabled).toEqual(true);
  });
  it('Should Not Simulate click events', () => {
    wrapper.find('button').simulate('click');
    expect(onButtonClick.callCount).toEqual(0);
  });
});



describe('<Button {...props} />', () => {
  let wrapper;
  let props = {
    backgroundColor: '#ec445c',
    color: '#ffffff',
    height: 45,
    width: '100%',
    buttonText: "Text Click",
    fontSize: 14,
    fontWeight: 'normal',
    marginTop: 10,
    marginBottom: 10,
    marginLeft: 10,
    marginRight: 10
  };
  beforeEach(() => wrapper = mount(<Button {...props} />));
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('Should Render a Button Container', () => {
    expect(wrapper.find('button[data-test="button-container"]').length).toEqual(1);
  });
  it('Should pass backgroundColor prop to Button as true', () => {
    expect(wrapper.find('[data-test="button-container"]').get(0).props.backgroundColor).toEqual(props.backgroundColor);
  });
  it('Should pass color prop to Button as true', () => {
    expect(wrapper.find('[data-test="button-container"]').get(0).props.color).toEqual(props.color);
  });
  it('Should pass height prop to Button as 45', () => {
    expect(wrapper.find('[data-test="button-container"]').get(0).props.height).toEqual(props.height);
  });
  it('Should pass width prop to Button as 100%', () => {
    expect(wrapper.find('[data-test="button-container"]').get(0).props.width).toEqual(props.width);
  });
  it('Should pass buttonText prop to Button as Test Click', () => {
    expect(wrapper.text()).toEqual(props.buttonText);
  });
  it('Should pass fontSize prop to Button as 14', () => {
    expect(wrapper.find('[data-test="button-container"]').get(0).props.fontSize).toEqual(props.fontSize);
  });
  it('Should pass fontWeight prop to Button as normal', () => {
    expect(wrapper.find('[data-test="button-container"]').get(0).props.fontWeight).toEqual(props.fontWeight);
  });
  it('Should pass marginTop prop to Button as 10', () => {
    expect(wrapper.find('[data-test="button-container"]').get(0).props.marginTop).toEqual(props.marginTop);
  });
  it('Should pass marginBottom prop to Button as 10', () => {
    expect(wrapper.find('[data-test="button-container"]').get(0).props.marginBottom).toEqual(props.marginBottom);
  });
  it('Should pass marginLeft prop to Button as 10', () => {
    expect(wrapper.find('[data-test="button-container"]').get(0).props.marginLeft).toEqual(props.marginLeft);
  });
  it('Should pass marginRight prop to Button as 10', () => {
    expect(wrapper.find('[data-test="button-container"]').get(0).props.marginRight).toEqual(props.marginRight);
  });
});

