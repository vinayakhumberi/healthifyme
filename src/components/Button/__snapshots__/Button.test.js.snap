// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `ReactWrapper {}`;

exports[`<Button {...props} /> renders correctly 1`] = `ReactWrapper {}`;

exports[`<Button isDisabled /> renders correctly 1`] = `ReactWrapper {}`;
